flink-core/pom.xml

```diff
@@ -146,6 +146,30 @@ under the License.
 				</configuration>
 			</plugin>
 
+			<!-- The following configuration is used to debug https://issues.apache.org/jira/browse/FLINK-31278.
+			Overriding the surefire config can be removed after FLINK-31278 is resolved. -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-surefire-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>default-tests</id>
+						<phase>test</phase>
+						<goals>
+							<goal>test</goal>
+						</goals>
+						<configuration>
+							<includes>
+								<include>${test.unit.pattern}</include>
+							</includes>
+							<argLine>${flink.surefire.baseArgLine} -Xmx${flink.XmxUnitTest} -XX:+HeapDumpOnOutOfMemoryError</argLine>
```

Comment:
ok, I don't know why it didn't cross my mind earlier (because I went over this code line as well), but we're setting this parameter through `JAVA_TOOL_OPTIONS` in [apache/flink:tools/ci/test_controller:67](https://github.com/apache/flink/blob/7e37d59f834bca805f5fbee99db87eb909d1814f/tools/ci/test_controller.sh#L67).